Rubberduck.Core/UI/CodeExplorer/Commands/ImportCommand.cs

```diff
@@ -180,6 +302,202 @@ protected virtual void ImportFiles(ICollection<string> filesToImport, IVBProject
+                {
+                    requiredBinaryNames.Add(filename, requiredBinaryFiles);
+                }
+            }
+
+            return requiredBinaryNames;
+        }
+
+        private ICollection<string> RequiredBinaryFiles(string filename)
+        {
+            var extension = Path.GetExtension(filename);
+            if (!ComponentTypesForExtension.TryGetValue(extension, out var componentTypes))
+            {
+                return new List<string>();
+            }
+
+            foreach (var componentType in componentTypes)
+            {
+                if (_binaryFileExtractors.TryGetValue(componentType, out var binaryExtractor))
+                {
+                    return binaryExtractor.RequiredBinaryFiles(filename, componentType);
+                }
+            }
+
+            return new List<string>();
+        }
+
+        private IDictionary<string, ICollection<string>> FilesWithoutRequiredBinaries(Dictionary<string, ICollection<string>> requiredBinaries)
+        {
+            var filesWithoutBinaries = new Dictionary<string, ICollection<string>>();
+            foreach (var (fileName, requiredBinariesForFile) in requiredBinaries)
+            {
+                var path = Path.GetDirectoryName(fileName);
+                var missingBinaries = requiredBinariesForFile
+                    .Where(binaryFileName => !_fileExistenceChecker.FileExists(Path.Combine(path, binaryFileName)))
+                    .ToList();
+
+                if (missingBinaries.Any())
+                {
+                    filesWithoutBinaries.Add(fileName, missingBinaries);
+                }
+            }
+
+            return filesWithoutBinaries;
+        }
+
+        private Dictionary<string, QualifiedModuleName> Modules(IDictionary<string, string> moduleNames, string projectId, DeclarationFinder finder)
+        {
+            var modules = new Dictionary<string, QualifiedModuleName>();
+            foreach (var (fileName, moduleName) in moduleNames)
+            {
+                var module = Module(moduleName, projectId, finder);
+                if (module.HasValue)
+                {
+                    modules.Add(fileName, module.Value);
+                }
+            }
+
+            return modules;
+        }
+
+        private bool ValuesAreUnique(Dictionary<string, string> moduleNames)
+        {
+            return moduleNames
+                .GroupBy(kvp => kvp.Value)
+                .All(moduleNameGroup => moduleNameGroup.Count() == 1);
+        }
+
+        private void NotifyUserAboutAbortDueToDuplicateComponent(IDictionary<string, string> moduleNames)
+        {
+            var firstDuplicateModuleName = moduleNames
+                .GroupBy(kvp => kvp.Value)
+                .First(moduleNameGroup => moduleNameGroup.Count() > 1)
+                .Key;
+            var message = string.Format(RubberduckUI.ImportCommand_DuplicateModule, firstDuplicateModuleName);
```

Comment:
this makes for a mediocre user experience. I'd want to know about all conflicts. That's something for a future PR though. A follow-up issue and a comment would be useful :)

Reply:
I have added a todo for it.